pkg/clusteragent/autoscaling/workload/patcher.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package workload
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload/model"
+	datadoghq "github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PatcherAdapter allows a workload patcher to access and modify data available in controller store
+type PatcherAdapter interface {
+	// GetPodAutoscalerFromOwnerRef returns the PodAutoscalerInternal object associated with the given owner reference
+	// NOTE: The returned PodAutoscalerInternal should never be modified
+	GetRecommendations(ns string, ownerRef metav1.OwnerReference) ([]datadoghq.DatadogPodAutoscalerContainerResources, error)
+}
+
+type patcherAdapter struct {
```

Comment:
Is it used anywhere ?

Reply:
Only the interface + the `New` function (or whatever returns the object, which is not yet plugged) needs to be public.

Reply:
We can leave it for later then